src/main.ts

```diff
@@ -89,10 +90,38 @@ function setupTrayItem() {
 
 function handleDownloads() {
     mainWindow.webContents.session.on('will-download', (event, item) => {
-        const savePath = path.join(
-            app.getPath('downloads'),
-            item.getFilename()
+        item.setSavePath(
+            getUniqueSavePath(item.getFilename(), app.getPath('downloads'))
         );
-        item.setSavePath(savePath);
     });
 }
+
+function getUniqueSavePath(filename: string, directory: string): string {
```

Comment:
Since I was reading this code, just for fun, I thought to see if I can simplify it. This is what I came up with:
```
     const [filenameWithoutExtension, extension] = filename.split('.');
     let n = 0;
     let uniquePath = path.join(directory, filename);
     while (!existsSync(uniquePath)) {
         n++;
         uniquePath = path.join(directory, [`${filenameWithoutExtension}(${n})`, extension].join('.'))
     }
     return uniquePath;
```

But this is not a blocking comment, just FYI, if you find something useful in this version use it otherwise your code is clear too no need to change.

We can merge once you do the split/join change.

Reply:
Ah wait, I just realized, my code has a bug! It'll split at any dot, not just the last dot.

Reply:
We can fix it by using the path module

```
const { name, ext } = path.parse('/home/user/dir/file.txt');
```

Reply:
we will also need to change the join too no  ?

```
> ["asd",undefined].join(".")
"asd." 
```
this is the joining with undefined adds a extra dot at the end

Reply:
Nice catch!

I don't see a corresponding name + extension join function in the path module. So we would need to filter out the undefined ourselves. `filter((x) => x)` might be the cleanest approach?

```
['test',undefined].filter((x) => x).join('-')
```

But please feel free to do something else if you have in mind too.